c10/util/ArrayRef.h

```diff
@@ -72,14 +72,12 @@ class ArrayRef final {
   constexpr ArrayRef(const T& OneElt) : Data(&OneElt), Length(1) {}
 
   /// Construct an ArrayRef from a pointer and length.
-  constexpr ArrayRef(const T* data, size_t length)
-      : Data(data), Length(length) {
+  ArrayRef(const T* data, size_t length) : Data(data), Length(length) {
```

Comment:
Hmm, can you please use one of convenience macros (like C10_HOST_CONSTEXPR_EXCEPT_WIN_CUDA ) rather than removing `constexpr` altogether.
Also, please leave a comment why this change is needed